runtime/vm/jfr.cpp

```diff
@@ -739,13 +752,39 @@ jfrSystemGC(J9HookInterface **hook, UDATA eventNum, void *eventData, void* userD
 	}
 }
 
+/**
+ * Hook for old garbage collection. Called without VM access.
+ *
+ * @param hook[in] the VM hook interface
+ * @param eventNum[in] the event number
+ * @param eventData[in] the event data
+ * @param userData[in] the registered user data
+ */
+static void
+jfrOldGarbageCollection(J9HookInterface **hook, UDATA eventNum, void *eventData, void* userData)
+{
+	MM_GCCycleEndEvent *event = (MM_GCCycleEndEvent *)eventData;
+	OMR_VMThread *omrVMThread = (OMR_VMThread *)event->omrVMThread;
```

Comment:
I think you need to a J9vmthread here `(J9VMThread *)omrVMThread->_language_vmthread`